06_timers/kmodule/timer_module.c

```diff
@@ -0,0 +1,114 @@
+// SPDX-License-Identifier: GPL-2.0 OR BSD-3-Clause
+
+#include <linux/init.h>
+#include <linux/module.h>
+#include <linux/kernel.h>
+#include <linux/printk.h>
+
+#include <linux/sysfs.h>
+#include <linux/device.h>
+
+#include <linux/types.h>
+#include <linux/slab.h>
+
+MODULE_AUTHOR("Andrii Synenko");
+MODULE_DESCRIPTION("List data module for Linux Kernel ProCamp");
+MODULE_LICENSE("Dual BSD/GPL");
+
+#define INTERVAL_1NS
+#define MS_TO_NS(x)     ((x) * NSEC_PER_MSEC)
+#define INTERVAL_1NS
+
+#define SYSFS_DIR_NAME	"timer_module"
+#define ABS_ENT_NAME	"time_of_last"
+#define REL_ENT_NAME	"time_since_last"
+
+static ssize_t abs_show(struct kobject *kobj, struct kobj_attribute *attr,
+					   char *buf)
+{
+	static u64 time_of_last;
+	u64 then_sec, then_ns;
+
+	then_sec = div64_u64_rem(time_of_last, NSEC_PER_SEC, &then_ns);
+
+	time_of_last = ktime_get_ns();
+
+	return sprintf(buf, "t[-1]: %llu.%03llu sec\n", then_sec, then_ns);
+}
+
+static ssize_t rel_show(struct kobject *kobj, struct kobj_attribute *attr,
+					   char *buf)
+{
+	static u64 time_of_last;
```

Comment:
These two static variables of name `time_of_last` (here and above) do not confuse compiler/linker but may confuse code supporters.

Reply:
Strange. But fixed

Reply:
It looks easy in a simple module with short functions, but in real life it is a place where you can make mistakes when modifying the code.